### PR TITLE
fix #13263 set ambiguousErrorMessages consistently across environments

### LINKED ISSUE #13263: Set ambiguousErrorMessages consistently across environments
See https://github.com/meteor/meteor/pull/12996

Whenever developers test and implement error handling locally, they expect that code will behave in the same way on production. Otherwise, a lot of time is wasted trying to understand different behaviour and failing to reproduce it locally.

cc @StorytellerCZ

### PR DESCRIPTION
**Pull request by mauroaccornero**
remove Meteor.isProduction from isErrorAmbiguous check in accounts_server.js

fixes: #13263

### CODE REVIEW

File: packages/accounts-base/accounts_server.js [hunk 1]
```diff
@@ -1519,11 +1519,11 @@ export class AccountsServer extends AccountsCommon {
   }
 
   _handleError = (msg, throwError = true, errorCode = 403) => {
-    const isErrorAmbiguous = this._options.ambiguousErrorMessages ?? Meteor.isProduction;
+    const isErrorAmbiguous = this._options.ambiguousErrorMessages;
```
Comment (StorytellerCZ) [inline packages/accounts-base/accounts_server.js]: The default for this needs to be `true` for security purposes.

**Review by StorytellerCZ — CHANGES_REQUESTED** [review]

File: packages/accounts-base/accounts_server.js (hunk 1 above)
Comment (mauroaccornero) [reply packages/accounts-base/accounts_server.js]: sure, thanks for the feedback!

**Review by StorytellerCZ — DISMISSED** [review]

**Comment by StorytellerCZ** [comment]
You will probably have to change the tests for this feature as well.

**Comment by mauroaccornero** [comment]
> You will probably have to change the tests for this feature as well.

it looks like it's something related to the docs in CircleCI (https://circleci.com/api/v1.1/project/github/meteor/meteor/141065/output/101/0?file=true&allocation-id=66c031f0bd2bca4b21b37bf0-0-build%2FABCDEFGH), all the tests are green (https://github.com/meteor/meteor/pull/13277/checks?check_run_id=28887631134)

edit: I think the issue is related to the docs inside CircleCi which is trying to clone the branch from this repository but this pr is from a fork on my GitHub account. Just for example, this one worked https://github.com/meteor/meteor/pull/13269/checks?check_run_id=28876123354

I'm not sure how to proceed with this.

**Comment by StorytellerCZ** [comment]
OK, don't worry about the CircleCI, check out the Travis one which I think runs the Tiny tests.

**Comment by mauroaccornero** [comment]
> OK, don't worry about the CircleCI, check out the Travis one which I think runs the Tiny tests.

Thanks for the hint, I've to update a couple of messages for the account password test.

**Review by StorytellerCZ — APPROVED** [review]

**Review by nachocodoner — APPROVED** [review]

**Comment by nachocodoner** [comment]
Docs tests fail when contributing from forks. We're aware of this and hope to fix it soon.

Your PR seems correct now. From now on, ambiguous error messages will be shown in both dev and prod modes, with full details only when specifically set.

**Pull request merged**